deno/lib/types.ts

```diff
@@ -855,6 +856,29 @@ export class ZodString extends ZodType<string, ZodStringDef, string> {
         input.data = input.data.toLowerCase();
       } else if (check.kind === "toUpperCase") {
         input.data = input.data.toUpperCase();
+      } else if (check.kind === "hostname") {
+        const domainNameRegex =
+          /^(?!-)(?!.*--)(?!.*\.\.)(?!.*\.$)[a-zA-Z0-9-]{1,63}(?<!-)(\.[a-zA-Z0-9-]{1,63})*$/;
+        const punycodeRegex = /^xn--[a-zA-Z0-9-]{1,63}$/;
```

Comment:
We still should check for the ending TLD for punycode, hostname should just not be able to accept `xn--d1acj3b` without the complete name like `xn--d1acj3b.com` or `xn--d1acj3b.org`

Reply:
fixed the issue and created a pr